src/ad/kandidatModal/LeggTilKandidatModal.tsx

```diff
@@ -46,14 +84,23 @@ class LeggTilKandidatModal extends React.Component {
         resetHentKandidatMedFnr();
     }
 
-    componentDidUpdate(prevProps) {
-        const { kandidatStatus } = this.props;
-        if (kandidatStatus !== prevProps.kandidatStatus) {
-            if (kandidatStatus === Hentestatus.SUCCESS) {
+    componentDidUpdate(prevProps: Props) {
+        const { kandidatStatus, kandidat, kandidatlisteStatus, kandidatliste } = this.props;
+
+        if (
+            kandidatStatus !== prevProps.kandidatStatus ||
+            kandidatlisteStatus !== prevProps.kandidatlisteStatus
+        ) {
+            if (
+                kandidatStatus === Hentestatus.SUCCESS &&
+                kandidatlisteStatus === Hentestatus.SUCCESS &&
+                kandidat &&
+                kandidatliste
```

Comment:
Dette er essensen. Hadde en bug der `kandidatliste` ikke var definert når man kjørte funksjonen `kandidatenFinnesAllerede`. Har lagt til både `kandidat` og `kandidatliste` som argumenter i denne slik at det blir tydeligere hva man trenger for å kjøre den.

i componentDidUpdate her vil vi nå sjekke om både kandidat OG kandidatlisten er definert før vi prøver å vise noe feilmeldinger eller vise noen kandidat.